pkg/pgclient/client.go

```diff
@@ -140,25 +150,34 @@ func NewClientWithPool(cfg *Config, numCopiers int, connPool *pgxpool.Pool, mt t
 		AsyncAcks:              cfg.AsyncAcks,
 	}
 
+	labelsReader := lreader.NewLabelsReader(dbConn, labelsCache)
+	exemplarKeyPosCache := cache.NewExemplarLabelsPosCache(cfg.CacheConfig)
+
+	dbQuerierConn := pgxconn.NewQueryLoggingPgxConn(connPool)
+	dbQuerier := querier.NewQuerier(dbQuerierConn, metricsCache, labelsReader, exemplarKeyPosCache, mt.ReadAuthorizer())
+	queryable := query.NewQueryable(dbQuerier, labelsReader)
+
 	var (
-		dbIngestor          *ingestor.DBIngestor
-		exemplarKeyPosCache = cache.NewExemplarLabelsPosCache(cfg.CacheConfig)
+		err             error
+		telemetryEngine telemetry.Engine
 	)
+	telemetryEngine, err = telemetry.NewEngine(dbConn, PromscaleID, queryable)
```

Comment:
I'd recommend keeping telemetry engine creation outside of Client (that reads and writes). Imho those are different concerns. It would also make it easier to test Client as we can just inject telemetry engine...

Reply:
But telemetry engine needs queryable and the client needs telemetry. So if I keep it outside, like in runner.go, it will be a egg-chicken problem.

Reply:
btw i agree, i should put it somewhere else